packages/editor/src/components/post-locked-modal/index.js

```diff
@@ -0,0 +1,230 @@
+		if ( isLocked || ! activePostLock ) {
+			return;
+		}
+
+		const data = {
+			action: 'wp-remove-post-lock',
+			_wpnonce: postLockUtils.unlockNonce,
+			post_ID: postId,
+			active_post_lock: activePostLock,
+		};
+
+		jQuery.post( {
+			async: false,
+			url: postLockUtils.ajaxUrl,
+			data,
+		} );
+	}
+
+	render() {
+		const { user, postId, isLocked, isTakeover, postLockUtils, postType } = this.props;
+		if ( ! isLocked ) {
+			return null;
+		}
+
+		const userDisplayName = user.name || __( 'Another user' );
+		const userAvatar = user.avatar;
+
+		const unlockUrl = addQueryArgs( 'post.php', {
+			'get-post-lock': '1',
+			lockKey: true,
+			post: postId,
+			action: 'edit',
+			_wpnonce: postLockUtils.nonce,
+		} );
+		const allPosts = getWPAdminURL( 'edit.php' );
+		return (
+			<Modal
+				title={ isTakeover ? __( 'Someone else has taken over this post.' ) : __( 'This post is already being edited.' ) }
+				focusOnMount={ true }
+				shouldCloseOnClickOutside={ false }
+				shouldCloseOnEsc={ false }
+				isDismissable={ false }
+				className="editor-post-locked-modal"
+			>
+				{ !! userAvatar && (
+					<img
+						src={ userAvatar }
+						alt={ __( 'Avatar' ) }
+						className="editor-post-locked-modal__avatar"
+					/>
+				) }
+				{ !! isTakeover && (
+					<div>
+						<div>
+							{ sprintf(
+								__( '%s now has editing control of this %s. Don\'t worry, your changes up to this moment have been saved' ),
+								userDisplayName,
+								postType
+							) }
+						</div>
+						<p>
+							<a href={ allPosts }>
+								{ __( 'View all posts' ) }
+							</a>
+						</p>
+					</div>
+				) }
+				{ ! isTakeover && (
+					<div>
+						<div>
+							{ sprintf(
+								__( '%s is currently working on this %s, which means you cannot make changes, unless you take over.' ),
+								userDisplayName,
+								postType
+							) }
+						</div>
+
+						<div className="editor-post-locked-modal__buttons">
+							<Button isDefault isLarge href={ allPosts }>
+								{ __( 'All Posts' ) }
+							</Button>
+							<PostPreviewButton />
+							<Button isPrimary isLarge href={ unlockUrl }>
+								{ __( 'Take Over' ) }
+							</Button>
+						</div>
+					</div>
+				) }
+			</Modal>
+		);
+	}
+}
+
+export default compose(
+	withSelect( ( select ) => {
+		const {
+			getEditorSettings,
+			isPostLocked,
+			isPostLockTakeover,
+			getPostLockUser,
+			getCurrentPostId,
+			getActivePostLock,
+			getEditedPostAttribute,
+		} = select( 'core/editor' );
+		return {
+			isLocked: isPostLocked(),
+			isTakeover: isPostLockTakeover(),
+			user: getPostLockUser(),
+			postId: getCurrentPostId(),
+			postLockUtils: getEditorSettings().postLockUtils,
+			activePostLock: getActivePostLock(),
+			postType: getEditedPostAttribute( 'type' ),
+		};
+	} ),
+	withDispatch( ( dispatch ) => {
+		const { autosave, updatePostLock } = dispatch( 'core/editor' );
+		return {
+			autosave,
+			updatePostLock,
+		};
+	} ),
+
```

Comment:
We're inconsistent with newlines between these members of the array. I'd recommend omitting them for broader consistency with the project.